service_configuration_lib/spark_config.py

```diff
@@ -334,7 +339,20 @@ def get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
         log.warning(
             f'\nSetting spark.dynamicAllocation.maxExecutors as {max_executors}. If you wish to '
             f'change the value of maximum executors, please provide the exact value of '
-            f'spark.dynamicAllocation.maxExecutors in --spark-args\n',
+            f'spark.dynamicAllocation.maxExecutors in your spark args\n',
+        )
+
+    if 'jupyterhub' in spark_app_name and 'spark.dynamicAllocation.initialExecutors' not in spark_opts:
```

Comment:
Is this the same logic in `paasta spark-run`?
Since we near end of year, would it better to make the change  in `spark-tools` to reduce the impact radius.
And then have a backlog ticket to have a common change in `service_configuration_lib` in early 2023?

Reply:
I didn't understand. The condition check here that `spark_app_name` has `jupyterhub` is already reducing the impact radius